src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -1586,6 +1590,44 @@ private void walk(IndexNode inode, Consumer<IndexNode> consumer) {
         return cen;
     }
 
+    /**
+     * Check Inode.name to see if it includes a "." or ".." in the name array
+     * @param path  the path as stored in Inode.name to verify
+     * @return true if the path contains a "." or ".." entry; false otherwise
+     */
+    private boolean hasDotOrDotDot(byte[] path) {
+        boolean dotOrDotDotFound = false;
+        // Inode.name always includes "/" in path[0]
+        if (path.length == 1) {
+            return false;
```

Comment:
It may be useful to add "assert path[0] == '/';" at the start of this method.

Reply:
I added it per your suggestion, though IndexNode(byte[] cen, int pos, int nlen) which is only used by initCen() will already guarantee the leading "/" is there